perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ConsentsManagerBlImpl.java

```diff
@@ -90,7 +95,20 @@ public Consent createConsent(PerunSession sess, Consent consent) throws ConsentE
 		Set<AttributeDefinition> filter = new HashSet<>();
 		for (Facility facility : consentHub.getFacilities()) {
 			for (Resource resource : perunBl.getUsersManagerBl().getAssignedResources(sess, facility, user)) {
+				Member member;
+				try {
+					Vo vo = getPerunBl().getVosManagerBl().getVoById(sess, resource.getVoId());
+					member = getPerunBl().getMembersManagerBl().getMemberByUser(sess, vo, user);
+				} catch (VoNotExistsException | MemberNotExistsException e) {
+					throw new InternalErrorException(e);
+				}
+				List<Group> groups = getPerunBl().getGroupsManagerBl().getAssignedGroupsToResource(sess, resource, member);
+				boolean isExpiredInResource = groups.stream()
+							.allMatch(group -> getPerunBl().getGroupsManagerBl().getDirectMemberGroupStatus(sess, member, group).equals(MemberGroupStatus.EXPIRED));
```

Comment:
```suggestion
							.allMatch(group -> getPerunBl().getGroupsManagerBl().getTotalMemberGroupStatus(sess, member, group).equals(MemberGroupStatus.EXPIRED));
```
The difference is that a member could have EXPIRED direct status and VALID indirect status through a subgroup. But totally, he is VALID. What do you think?

Reply:
You're right, I wasn't sure which one to use.